evolver/calibration/actions.py

```diff
@@ -2,23 +2,37 @@
 from copy import deepcopy
 from abc import ABC, abstractmethod
 
+import pydantic
+
 
 class CalibrationAction(ABC):
+    class UserInput(pydantic.BaseModel): ...
+
     @abstractmethod
     def execute(self, state: Dict[str, Any], payload: Dict[str, Any] = None) -> Dict[str, Any]:
         pass
 
 
 class DisplayInstructionAction(CalibrationAction):
+    class UserInput(pydantic.BaseModel):
+        confirmation: bool = pydantic.Field(..., description="Confirm that the instruction has been completed")
+
     def __init__(self, description: str, name: str):
         self.description = description
         self.name = name
 
     def execute(self, state: Dict[str, Any], payload: Dict[str, Any] = None) -> Dict[str, Any]:
+        user_input = self.UserInput(**payload)
```

Comment:
I wonder if the input type should actually be the already deserialized `UserInput`, this could be done upper layers by the framework (maybe in `calibrationProcedure.dispatch` , since then it knows the action)? In other words the API becomes:

```py
def execute(self, state: Dict[str, Any], payload: UserInput = None) -> ...
```

Reply:
Nice idea, will implement.